RFXtrx/__init__.py

```diff
@@ -770,8 +770,8 @@ def receive_blocking(self):
             if not data or data == '\x00':
                 continue
             pkt = bytearray(data)
-            while len(pkt) < pkt[0]:
-                data = self.sock.recv(pkt[0])
+            while len(pkt) < pkt[0]+1:
```

Comment:
Why is there a +1 here? Feel like if that has been an issue nothing aught to have worked?

Reply:
`pkt[0]` contains the length of the _rest_ of the packet (not including the length byte itself).
So the actual length of the full packet is pkt[0]+1.

I think the reason this worked uphill now is that the packet data so far has been so short that `sock.recv(pkt[0])` always got the full packet in one go. This is the one case where the while loop is correct, since `len(pkt)==1` at that point.